django/db/models/base.py

```diff
@@ -836,7 +836,11 @@ def delete(self, using=None):
 
         collector = Collector(using=using)
         collector.collect([self])
-        collector.delete()
+        deleted, _rows_count = collector.delete()
+        if rows_count:
```

Comment:
A parameter the controls whether a tuple or a single item is returned from a method doesn't seem like a great API. I'd be inclined to drop `rows_count` unless you discussed this with someone else.